File: src/2025-03-04-03-28-40--deepseek-chat-0304/javascript/exercises/practice/pig-latin/pig-latin.js
const isVowel = (char) => /[aeiou]/i.test(char);

const startsWithVowelLike = (word) => 
  isVowel(word[0]) || word.startsWith('xr') || word.startsWith('yt');

const getConsonantCluster = (word) => {
  // Handle 'qu' cases
  const quIndex = word.indexOf('qu');
  if (quIndex !== -1) {
    return word.slice(0, quIndex + 2);
  }
  
  // Handle 'y' as consonant at start
  if (word[0] === 'y') {
    return 'y';
  }
  
  // Handle 'y' after consonants
  const yIndex = word.indexOf('y');
  if (yIndex > 0) {
    return word.slice(0, yIndex);
  }
  
  // Get all consonants at start, treating 'th' as a cluster
  let i = 0;
  while (i < word.length && !isVowel(word[i]) && word[i] !== 'y') {
    // Handle 'th' as a single cluster
    if (word[i] === 't' && word[i+1] === 'h') {
      i += 2;
      break;
    }
    i++;
  }
  return word.slice(0, i);
};

export const translate = (text) => {
  return text.split(' ').map(word => {
    if (startsWithVowelLike(word)) {
      return word + 'ay';
    }
    
    const cluster = getConsonantCluster(word);
    const rest = word.slice(cluster.length);
    return rest + cluster + 'ay';
  }).join(' ');
};

File: src/2025-03-04-03-28-40--deepseek-chat-0304/python/exercises/practice/pig-latin/pig_latin.py
def translate(text):
    vowels = {'a', 'e', 'i', 'o', 'u'}
    result = []
    
    for word in text.split():
        # Rule 1: Starts with vowel or 'xr' or 'yt'
        if word[0] in vowels or word.startswith('xr') or word.startswith('yt'):
            result.append(word + 'ay')
            continue
            
        # Find the first vowel or 'y' position
        vowel_pos = 0
        while vowel_pos < len(word):
            if word[vowel_pos] in vowels or (vowel_pos > 0 and word[vowel_pos] == 'y'):
                break
            vowel_pos += 1
            
        # Rule 3: Handle 'qu' cases
        if vowel_pos > 0 and word[vowel_pos] == 'u' and word[vowel_pos-1] == 'q':
            vowel_pos += 1
            
        # Rule 2 & 4: Move consonants to end
        translated = word[vowel_pos:] + word[:vowel_pos] + 'ay'
        result.append(translated)
        
    return ' '.join(result)
